hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -360,6 +383,101 @@ private static Option<HoodieIndexMetadata> readIndexDefFromStorage(
     }
   }
 
+  /**
+   * Gets the latest table schema using TableSchemaResolver.
+   *
+   * @param tableConfig The table configuration
+   * @return Option containing the latest table schema, or empty if schema cannot be resolved
+   */
+  Option<Schema> getLatestTableSchema(HoodieTableConfig tableConfig) {
+    try {
+      TableSchemaResolver schemaResolver = new TableSchemaResolver(this);
+      // Get schema without metadata fields for column type checking
+      return Option.of(schemaResolver.getTableAvroSchema(false));
+    } catch (Exception e) {
+      LOG.warn("Failed to resolve latest table schema, falling back to table create schema", e);
+      // Fallback to table create schema if schema resolution fails
+      return tableConfig.getTableCreateSchema();
+    }
+  }
+
+  static HoodieIndexMetadata disableV1ColumnStatsForTimestampMillisColumns(HoodieIndexMetadata originalMetadata,
+                                                                           List<String> timestampMillisColumns) {
+    Map<String, HoodieIndexDefinition> indexDefinitions = originalMetadata.getIndexDefinitions();
+    Map<String, HoodieIndexDefinition> filteredIndexDefinitions = new HashMap<>();
+
+    for (Map.Entry<String, HoodieIndexDefinition> entry : indexDefinitions.entrySet()) {
+      String indexName = entry.getKey();
+      HoodieIndexDefinition indexDef = entry.getValue();
+      // Only check V1 column stats index.
+      if (indexDef.getVersion() == HoodieIndexVersion.V1
+          && HoodieTableMetadataUtil.PARTITION_NAME_COLUMN_STATS.equals(indexName)) {
+        // Filter out timestamp_millis columns from sourceFields
+        List<String> sourceFields = indexDef.getSourceFields();
+        if (sourceFields != null && !sourceFields.isEmpty()) {
+          List<String> filteredSourceFields = sourceFields.stream()
```

Comment:
we have to parse nested fields as well. currently, we are only polling root level.

Reply:
This is fixed.